### ISSUE TITLE
VKB sticks if you open a menu

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#5480]
Press a key
Open a menu
Voila, no way to release other than down/up again
Tricky....

**Comment by baconpaul** [comment]
OK here's how it is now

Press key
Release key
Works

Press Key
Open Menu
Close Menu
Release key
Works

Press Key
Open Menu
Release Key
Close Menu
Note sticks

Press Key
Open Menu
Release Key
Close Menu
Press Key (nothing happens)
Release Key
Note Releases

So that middle case is really the hard one

May be too hard to fix also. Debating. And not entirely clear what the fix is to me right now.

**Comment by mkruselj** [comment]
Another possibly related behavior:

1. Enable keyboard shortcuts
2. Press Alt/Option and hold it
3. Press K and shortly hold it - VKB opens
4. Right after you pressed K release Alt/Option while K is still held - a note is played matching the K key

I would expect keys would only play when actually pressing them, not when coming from a modifier key release.

**Comment by baconpaul** [comment]
Yeah so that’s because the widget scans the keyboard when there’s a change which is why chording and stuff works. And releasing alt makes it scan and see k is down. The midi keyboard is not a press api it is a scan on change api (roughly)

my guess is all of these will be very very hard to fix

**Comment by mkruselj** [comment]
😢

**Comment by baconpaul** [comment]
alt k doesn't work on macOS either
it and alt t both just give beeps
so the keyboard mapping is somehow not right even before we fix this.
hmm.

**Comment by mkruselj** [comment]
@baconpaul There is a potential workaround for this issue. Alt+K is the current shortcut (because K=keyboard). We could in theory modify it to be Alt+V (V=virtual keyboard?), perhaps. It's not very intuitive but it would sidestep this issue. Hmhmhmhm...
